src/Microsoft.AspNetCore.Server.Kestrel/Http/Frame.cs

```diff
@@ -804,8 +804,43 @@ protected bool TakeStartLine(SocketInput input)
                 QueryString = queryString;
                 HttpVersion = httpVersion;
 
-                bool caseMatches;
+                if (requestUrlPath.Length > 0 && requestUrlPath[0] != '/')
+                {
+                    int hostIndex;
+                    if (requestUrlPath.StartsWith("http://"))
```

Comment:
I'd be interested in the allocation profile on this. Just curious if we could just operate on the bytes before materializing to a string or all the operations here non-allocating?